pep-0692.rst

```diff
@@ -0,0 +1,438 @@
+        name: str
+        year: int
+    
+    def foo(**kwargs: Movie) -> None: ...
+
+means that each keyword argument in ``foo`` is itself a ``Movie`` dictionary
+that has a ``name`` key with a string type value and a ``year`` key with an
+integer type value. Therefore, in order to support specifying ``kwargs`` type
+as a ``TypedDict`` without breaking current behaviour, a new syntax has to be
+introduced.
+
+Specification
+=============
+
+To support the aforementioned use case we propose to use the double asterisk
+syntax inside of the type annotation. The required grammar change is discussed
+in more detail in section `Grammar Changes`_. Continuing the previous example:
+
+::
+
+    def foo(**kwargs: **Movie) -> None: ...
+
+would mean that the ``**kwargs`` comprise two keyword arguments specified by
+``Movie`` (i.e. a ``name`` keyword of type ``str`` and a ``year`` keyword of
+type ``int``). Then, inside the function itself, the type checkers should treat
+the ``kwargs`` parameter as a ``TypedDict``:
+
+::
+
+    def foo(**kwargs: **Movie) -> None:
+        assert_type(kwargs, Movie)  # OK!
+
+
+Using the new annotation will not have any runtime effect - it should only be
+taken into account by the type checkers.
+
+Function calls with standard dictionaries
+-----------------------------------------
+
+Calling a function that has ``**kwargs`` typed using the ``**kwargs: **Movie``
+syntax with a standard dictionary must generate an error. For example:
+
+::
+
+    def foo(**kwargs: **Movie) -> None: ...
+
+    movie: dict[str, object] = {"name": "Life of Brian", "year": 1979}
+    foo(**movie)  # WRONG! Movie is of type dict[str, object]
+
+    typedMovie: Movie = {"name": "The Meaning of Life", "year": 1983}
+    foo(**typedMovie)  # OK!
+
+
+Keyword collisions
+------------------
+
+A ``TypedDict`` that is used to type ``**kwargs`` could potentially contain keys
+that are already defined in the function's signature. If the duplicate name is a
+standard argument, an error should be reported. If the duplicate name is a
+positional only argument, no errors should be generated. For example:
+
+::
+
+    def foo(name, **kwargs: **Movie) -> None: ...     # WRONG! "name" will
+                                                      # always bind to the first
+                                                      # parameter.
+
+    def foo(name, /, **kwargs: **Movie) -> None: ...  # OK! "name" is a
+                                                      # positional argument,
+                                                      # so **kwargs can contain
+                                                      # a "name" keyword.
+
+
+Required and non-required keys
+------------------------------
+
+By default all keys in a ``TypedDict`` are required. This behaviour can be
+overridden by setting the dictionary's ``total`` parameter as ``False``.
+Moreover, PEP 655 introduced new type qualifiers - ``typing.Required`` and
```

Comment:
```suggestion
Moreover, :pep:`655` introduced new type qualifiers - ``typing.Required`` and
```
Suggest making the PEP link explicit. (I can't remember if regular text in format "PEP ###" is auto-linked or not.)